lib/internal/bootstrap/pre_execution.js

```diff
@@ -168,6 +169,53 @@ function setupSignalHandlers() {
   process.on('removeListener', stopListeningIfSignal);
 }
 
+
+function setupStacktracePrinterOnSigint() {
+  if (!getOptionValue('--trace-sigint')) {
+    return;
+  }
+  const { SigintWatchdog, SignalFlags } = require('internal/watchdog');
+
+  const watchdog = new SigintWatchdog();
+  watchdog.oninterrupt = function oninterrupt(flag) {
+    if (process.listenerCount('SIGINT') > 0) {
+      // Simply ignore this watchdog interrupt to prevent duplicate
+      // handling of SIGINT.
+      return;
+    }
+    const console = require('console');
+
+    // Keyboard interruption may not be captured by uncaughtException
+    // listener.
+    // eslint-disable-next-line no-restricted-syntax
+    const err = new Error('Script execution was interrupted by `SIGINT`');
+    Object.defineProperty(err, 'name', {
+      value: 'KEYBOARD_INTERRUPT',
+      enumerable: false,
+      writable: true,
+      configurable: true
+    });
+    if (flag === SignalFlags.FromIdle) {
+      const stackTraceLimit = Error.stackTraceLimit;
+      Error.stackTraceLimit = 0;
+      // eslint-disable-next-line no-restricted-syntax
+      Error.captureStackTrace(err, oninterrupt);
+      Error.stackTraceLimit = stackTraceLimit;
+      // util.inspect would wrap no stacktrace error with brackets.
+      console.error(`${err.name}: ${err.message}`);
+    } else {
+      // eslint-disable-next-line no-restricted-syntax
+      Error.captureStackTrace(err, oninterrupt);
+      console.error(err);
+    }
+    // Re-raising SIGINT.
+    watchdog.stop();
+    process.kill(process.pid, 'SIGINT');
+  };
+  watchdog.start();
+  return watchdog;
```

Comment:
nit: I believe we are not using the return value anywhere in the code. We can get rid of this line.